Kontent.Ai.AspNetCore/Webhooks/Models/WebhookNotification.cs

```diff
@@ -0,0 +1,138 @@
+    {
+        /// <summary>
+        /// A collection of webhook notifications for each modified object.
+        /// </summary>
+        [JsonPropertyName("notifications"), JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+        public WebhookModel[] Notifications { get; set; }
+    }
+
+    /// <summary>
+    /// The webhook model that contains data and message.
+    /// </summary>
+    public class WebhookModel
+    {
+        /// <summary>
+        /// Data relevant to the object that triggered the webhook.
+        /// </summary>
+        [JsonPropertyName("data"), JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+        public WebhookData Data { get; set; }
+        /// <summary>
+        /// The Message object contains information about the origin of the notification.
+        /// </summary>
+        [JsonPropertyName("message"), JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+        public WebhookMessage Message { get; set; }
+    }
+
+    /// <summary>
+    /// Data relevant to the object that triggered the webhook.
+    /// </summary>
+    public class WebhookData
+    {
+        /// <summary>
+        /// Metadata of the modified object.
+        /// </summary>
+        [JsonPropertyName("system"), JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+        public WebhookItem System { get; set; }
+    }
+
+    /// <summary>
+    /// The Message object contains information about the origin of the notification.
+    /// </summary>
+    public class WebhookMessage
+    {
+        /// <summary>
+        /// Identifier of the webhook.
+        /// </summary>
+        [JsonPropertyName("environment_id")]
+        public Guid EnvironmentId { get; set; }
+
+        /// <summary>
+        /// Type of the object that triggered the webhook (content_item_variant, taxonomy, ...)
+        /// </summary>
+        [JsonPropertyName("object_type")]
+        public string ObjectType { get; set; }
+
+        /// <summary>
+        /// Codename of the action that triggered the webhook (published, unpublished, created, deleted, ...).
+        /// </summary>
+        [JsonPropertyName("action")]
+        public string Action { get; set; }
+
+        /// <summary>
+        /// Codename of the delivery slot where the webhook was triggered (preview, published).
+        /// </summary>
+        [JsonPropertyName("delivery_slot")]
+        public string DeliverySlot { get; set; }
+    }
+
+
+    /// <summary>
+    /// Represents metadata of the modified object.
+    /// </summary>
+    public class WebhookItem
+    {
+        /// <summary>
+        /// The object's ID.
+        /// </summary>
+        [JsonPropertyName("id"), JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+        public Guid? Id { get; set; }
+
+        /// <summary>
+        /// The item's name.
+        /// </summary>
+        [JsonPropertyName("name"), JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
```

Comment:
are those `JsonIgnoreCondition` correct? 

from documentation
WhenWritingNull - The property is ignored on serialization if it's a reference type null, or a nullable value type null.

this property is not even nullable or am I missing something?

https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/ignore-properties

Reply:
reference types are always nullable, `?` added where applicable. I'm not 100% sure about the decorators intended purpose but I followed the same approach as was used previously.